### PR TITLE
Permission fixes

### PR DESCRIPTION
**Pull request by rathorevaibhav**
Targets https://github.com/ColoredCow/portal/issues/1364

### Description
1. All modules seeder files for permissions have been fixed - along with permission granting to the user roles
2. Roles table cleaned up with less default roles
3. navbar.blade fixed with correct permissions checks
4. Some files have been deleted that were redundant
5. UI cleanups have been done on user settings and role settings page
6. In the user management screen, added check that a user cannot delete themself
7. Showing count of permissions assigned to a role in the role listing screen
8. Home page (dashboard) update for users with no role. They will see a nice message to request administration for access.

### Checklist:

- [x] I have performed a self-review of my own code.

### Todo
1. This PR only covers hiding CTAs and menu options if user does not have permission. Adding checks when user accesses certain URLs directly will be worked on in a separate PR.

### CODE REVIEW

File: Modules/HR/Database/Seeders/HRDatabaseSeeder.php [hunk 1]
```diff
@@ -16,8 +16,6 @@ public function run()
     {
         Model::unguard();
         $this->call(HRPermissionsTableSeeder::class);
-        $this->call(HRRoleHasPermissionsTableSeeder::class);
-        $this->call(VolunteersModuleSeeder::class);
```
[the remark this answers is not in the document]
Comment (P4NK4J) [reply Modules/HR/Database/Seeders/HRDatabaseSeeder.php:20]: Ah! I got to know they are deleted below, resolving the comment.

File: Modules/User/Resources/assets/js/components/UpdateUserRoleModal.vue [hunk 2]
```diff
@@ -31,13 +29,13 @@
                                         :data-label="role.label"
                                         :value="role.id"
                                         @click="checkForSuperAdmin(role)"
-                                        class=""> {{ role.label }}
+                                    />
+									{{ role.label }}
```
Comment (P4NK4J) [inline Modules/User/Resources/assets/js/components/UpdateUserRoleModal.vue:33]: The indentation here is quite off, can we do something to bring it in alignment ?

File: Modules/User/Resources/assets/js/components/UserListing.vue [hunk 3]
```diff
@@ -14,14 +14,17 @@
...
-                            <div style="width:30px;" class="mr-2">
+                        <span class="align-items-start d-flex justify-content-start">
+                            <div class="mr-2 w-30">
                                 <img style="border-radius:50%"  class="w-full" :src="user.avatar" alt="">
                             </div>
-                            {{ user.name }}
+							<div>
+								<div>{{ user.name }}</div>
+								<div class="text-secondary fz-14">{{ user.email }}</div>
+							</div>
```
Comment (P4NK4J) [inline Modules/User/Resources/assets/js/components/UserListing.vue:24]: Indentation is quite off

**Review by P4NK4J — CHANGES_REQUESTED** [review]
Overall it looks good. I have left some comments, please check

**Review by GauravGusain98 — APPROVED** [review]
Looks fine just fix the issues mentioned by PK

File: Modules/User/Resources/assets/js/components/UpdateUserRoleModal.vue (hunk 2 above)
Comment (rathorevaibhav) [reply Modules/User/Resources/assets/js/components/UpdateUserRoleModal.vue:33]: It's aligned @P4NK4J 
The `<input />` and `{{role.label}}` are child of same parent. Hence they are at same indentation level. Let me know if it should be something else

File: Modules/User/Resources/assets/js/components/UserListing.vue (hunk 3 above)
Comment (rathorevaibhav) [reply Modules/User/Resources/assets/js/components/UserListing.vue:24]: Sorry, I don't understand where the indentation is off here. Can you share a bit more?
This seems just alright to me:
```vue
<div>
    <div>{{ user.name }}</div>
    <div class="text-secondary fz-14">{{ user.email }}</div>
</div>
```

File: Modules/HR/Database/Seeders/HrApplicantsTableSeeder.php [hunk 4]
```diff
@@ -14,8 +14,10 @@ class HrApplicantsTableSeeder extends Seeder
      */
     public function run()
     {
-        Applicant::factory()
-            ->count(2)
-            ->create();
+        if (! app()->environment('production')) {
```
[the remark this answers is not in the document]
Comment (rathorevaibhav) [reply Modules/HR/Database/Seeders/HrApplicantsTableSeeder.php:17]: You mean the main module seeder file?

File: Modules/AppointmentSlots/Database/Seeders/AppointmentSlotsDatabaseSeeder.php [hunk 5]
```diff
@@ -16,6 +16,5 @@ public function run()
     {
         Model::unguard();
         $this->call(AppointmentSlotsPermissionsTableSeeder::class);
-        $this->call(AppointmentSlotsRoleHasPermissionsTableSeeder::class);
```
[the remark this answers is not in the document]
Comment (rathorevaibhav) [reply Modules/AppointmentSlots/Database/Seeders/AppointmentSlotsDatabaseSeeder.php:19]: the role & permission assignment handling are now cleaned and included in permission table seeder itself. So we don't need this redundant role-permission seeder file

File: Modules/HR/Database/Seeders/HrApplicantsTableSeeder.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (rathorevaibhav) [reply Modules/HR/Database/Seeders/HrApplicantsTableSeeder.php:17]: @P4NK4J I was thinking about it but then I realized there maybe extra commands in the individual seeder files which may not be around fake data. For example this file: `HRJobsSeederTableSeeder.php`. It has Artisan call not dependent on environment.

Hence I thought to keep the env check as it is within the individual table seeder files. Let me know what you think.

File: Modules/User/Resources/assets/js/components/UserListing.vue (hunk 3 above)
Comment (rathorevaibhav) [reply Modules/User/Resources/assets/js/components/UserListing.vue:24]: Got it. The `</td>` tag was not aligned. Fixed it.

File: Modules/HR/Database/Seeders/HrApplicantsTableSeeder.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (P4NK4J) [reply Modules/HR/Database/Seeders/HrApplicantsTableSeeder.php:17]: Okay, that sounds good.

**Review by P4NK4J — APPROVED** [review]
Looks good to me

**Pull request merged**